Apps/W1/EDocumentConnectors/SignUp/app/src/Helpers.Codeunit.al

```diff
@@ -0,0 +1,106 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.EServices.EDocumentConnector.SignUp;
+
+using System.Reflection;
+using Microsoft.Utilities;
+using System.Integration;
+using Microsoft.eServices.EDocument;
+
+codeunit 6385 Helpers
+{
+    Access = Internal;
+
+    var
+        ClaimTypeTxt: Label 'exp', Locked = true;
+
+    #region public methods
+
+    [NonDebuggable]
+    procedure ParseJsonString(HttpContent: HttpContent): Text
+    var
+        JsonObject: JsonObject;
+        Content: Text;
+    begin
+        if not HttpContent.ReadAs(Content) then
+            exit;
+
+        if JsonObject.ReadFrom(Content) then
+            exit(Content);
+    end;
+
+    [NonDebuggable]
+    procedure GetJsonValueFromText(JsonText: Text; Path: Text): Text
+    var
+        JsonObject: JsonObject;
+        JsonToken: JsonToken;
+    begin
+        if JsonObject.ReadFrom(JsonText) then
+            if JsonObject.SelectToken(Path, JsonToken) then
+                exit(this.GetJsonValue(JsonToken.AsValue()));
+    end;
+
+    procedure IsTokenValid(InToken: SecretText): Boolean
+    begin
+        exit(this.GetTokenDateTimeValue(InToken, this.ClaimTypeTxt) > CurrentDateTime());
+    end;
+
+    procedure IsExFlowEInvoicing(EDocumentServiceCodeFilter: Text): Boolean
+    var
+        EDocumentService: Record "E-Document Service";
+    begin
+        if EDocumentServiceCodeFilter = '' then
+            exit;
+
+
+        EDocumentService.SetFilter(Code, EDocumentServiceCodeFilter);
+        EDocumentService.SetRange("Service Integration V2", EDocumentService."Service Integration V2"::"ExFlow E-Invoicing");
+        exit(not EDocumentService.IsEmpty());
+    end;
+
+    #endregion
+
+    #region local methods
+
+    local procedure GetTokenDateTimeValue(InToken: SecretText; ClaimType: Text): DateTime
+    var
+        TypeHelper: Codeunit "Type Helper";
+        Timestamp: Decimal;
+    begin
+        if Evaluate(Timestamp, this.GetValueFromToken(InToken, ClaimType)) then
+            exit(TypeHelper.EvaluateUnixTimestamp(Timestamp));
```

Comment:
Since BC25 we now have this feature for UnixTimestamp Evaluation in the System App.
New code should move away from the procedure in the TypeHelper Codeunit.
[EvaluateTimestamp](https://github.com/microsoft/BCApps/blob/main/src/System%20Application/App/Date%20and%20Time/src/UnixTimestamp.Codeunit.al)